frameworks/hdfs/universe/config.json

```diff
@@ -875,9 +890,14 @@
         },
         "hadoop_opts": {
           "type": "string",
-          "description": "Extra Java runtime options.",
+          "description": "Extra Java runtime options for all Hadoop nodes.",
           "default": ""
         },
+        "hadoop_heapsize": {
+          "type": "string",
```

Comment:
Don't we generally use integers for memory values?

Reply:
Ah generally yeah. Earlier versions of Hadoop require you to specify the multiplier type as well as the number eg `1024MB` instead of just `1024`. More recent versions, such as the one here, don't require this anymore so we can change this to a `number` type.

Reply:
One last question. Is this kind of change safe across upgrades?

Reply:
In this case, yeah it's safe because Hadoop already implicitly takes this setting into account by setting this value by default to 1000MB. Explicitly setting it doesn't change the functionality, and even upgrading to a package that has this setting with a different number simply changes the heap size available to the Hadoop nodes, so not exactly a breaking change.

Reply:
I mean that if one does a `dcos hdfs describe > config.json`, make an edit and then `dcos hdfs update start --options=config.json --version=next-version` where `next-version` is the one that now has an integer in this field and the one that was running had a string.

Reply:
Ah, well `hadoop_heapsize` doesn't currently exist at all in `config.json` so the first version of the package would have this field as an `integer` to begin with.

Reply:
(oh right ... it's been a long day :) )